apisix/discovery/eureka.lua

```diff
@@ -203,16 +205,18 @@ local function fetch_full_registry(premature)
                     nodes = core.table.new(#app.instance, 0)
                     up_apps[app.name] = nodes
                 end
+                local weight = default_weight
+                if metadata and metadata.weight then
+                    weight = tonumber(metadata.weight) or default_weight
+                else
+                    metadata.weight = nil
```

Comment:
why we need this line? Can we remove this line?

Reply:
and we need test case to confirm this change is correct.

Reply:
> why we need this line? Can we remove this line?

Which line ？

Reply:
`metadata.weight = nil`

Reply:
> and we need test case to confirm this change is correct.

We need to learn about the test framework. We haven't touched it before. We will submit the test case later.
haha

Reply:
> `metadata.weight = nil`
   remove useless data